### ISSUE TITLE
Updating existing vnet dns_server with Azure firewall leads to azurerm_virtual_network error.

### ISSUE THREAD

**Issue by LaurentLesle** [Azure/terraform-azurerm-caf-enterprise-scale#1136]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Versions

**terraform**: 1.9.5

**azure provider**: 3.107.0

**module**: 6.1.0

#### Describe the bug

When:

1. using the version of the module 6.1.0 and pinning the azurerm provider "~> 3.107.0"
2. Updating a virtual network DNS to set the private IP of the Azure firewall

```
╷
│ Error: ID was missing the `networkSecurityGroups` element
│ 
│   with azurerm_virtual_network.some_network["test"],
│   on main.tf line 132, in resource "azurerm_virtual_network" "some_network":
│  132: resource "azurerm_virtual_network" "some_network" {
│ 
╵
```

This bug has been reported in the azurerm provider
https://github.com/hashicorp/terraform-provider-azurerm/issues/26276#issue-2342071482

Therefore I propose to set the minimum version of the module to 3.108.0

From terraform.tf
```hcl
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~> 3.107"
      configuration_aliases = [
        azurerm.connectivity,
        azurerm.management,
      ]
    }
```

#### Steps to Reproduce

1. step 1
4. step 2
5. you get it...

**Comment by LaurentLesle** [comment]
@matt-FFFFFF once approved it would be good if you can release a 6.2.0 to set the minimum version of Azurerm to 3.108.0

**Comment by matt-FFFFFF** [comment]
Released in v6.2.0
